### PR TITLE
document setup for Monad Lib

### PR DESCRIPTION
**Pull request by lsf37**
- add document setup in preparation for AFP entry
- consequently clean up structure and document issues

### CODE REVIEW

File: lib/Monads/nondet/Nondet_While_Loop_Rules.thy [hunk 1]
```diff
@@ -618,7 +618,7 @@ lemma whileLoopE_liftE:
    apply (rule set_eqI, rule iffI)
     apply clarsimp
     apply (clarsimp simp: in_liftE whileLoop_def)
-    \<comment> \<open>The schematic existential is instantiated by 'subst isr_Inr_proj' ... 'rule refl' in two lines\<close>
+    \<comment> \<open>The schematic existential is instantiated by @{text "subst isr_Inr_proj' ... rule refl"} in two lines\<close>
```
Comment (corlewis) [inline lib/Monads/nondet/Nondet_While_Loop_Rules.thy:621]: My fault, but this should be `isr_Inr_projr`

**Review by michaelmcinerney — APPROVED** [review]
👍

**Review by corlewis — APPROVED** [review]
Looks good to me, except for a small typo in the first commit message; "the the repo history"

**Comment by lsf37** [comment]
> Looks good to me, except for a small typo in the first commit message; "the the repo history"

Thanks, well spotted and now fixed.

File: lib/Monads/nondet/Nondet_While_Loop_Rules.thy [hunk 2]
```diff
@@ -618,7 +618,7 @@ lemma whileLoopE_liftE:
    apply (rule set_eqI, rule iffI)
     apply clarsimp
     apply (clarsimp simp: in_liftE whileLoop_def)
-    \<comment> \<open>The schematic existential is instantiated by 'subst isr_Inr_proj' ... 'rule refl' in two lines\<close>
+    \<comment> \<open>The schematic existential is instantiated by @{text "subst isr_Inr_projr' ... rule refl"} in two lines\<close>
```
Comment (corlewis) [inline lib/Monads/nondet/Nondet_While_Loop_Rules.thy:621]: The `'` can be removed as well, it's left over from how the comment used to be formatted.
```suggestion
    \<comment> \<open>The schematic existential is instantiated by @{text "subst isr_Inr_projr ... rule refl"} in two lines\<close>
```

Comment (lsf37) [reply lib/Monads/nondet/Nondet_While_Loop_Rules.thy:621]: ah, yes

**Pull request merged**